### PR TITLE
msft: bump jdk17 to 17.0.3 and add aarch64 support to jdk11

### PR DESCRIPTION
**Pull request by gdams**
**Description:**
Describe your changes.

**Related issue:**
Add link to the related issue.

**Check list:**
- [ ] Mark if documentation changes are required.
- [ ] Mark if tests were added or updated to cover the changes.

### CODE REVIEW

File: dist/setup/index.js [hunk 1]
```diff
@@ -101717,21 +101717,24 @@ class MicrosoftDistributions extends base_installer_1.JavaBase {
             // TODO get these dynamically!
             // We will need Microsoft to add an endpoint where we can query for versions.
             const jdkVersions = [
+                {
+                    version: [17, 0, 3]
```
Comment (karianna) [inline dist/setup/index.js:101721]: Should 17.0.3 not replace 17.0.1.12.1?

Does 17.0.3 need to have the full build number written out?

File: dist/setup/index.js [hunk 2]
```diff
@@ -101717,21 +101717,24 @@ class MicrosoftDistributions extends base_installer_1.JavaBase {
...
+                },
                 {
                     version: [17, 0, 1, 12, 1]
                 },
                 {
                     version: [16, 0, 2, 7, 1]
+                },
+                {
+                    version: [11, 0, 15]
```
Comment (karianna) [inline dist/setup/index.js:101730]: Does 11.0.15 need to have the full version written out?

File: dist/setup/index.js [hunk 3]
```diff
@@ -101717,21 +101717,24 @@ class MicrosoftDistributions extends base_installer_1.JavaBase {
...
                 }
             ];
             // M1 is only supported for Java 16 & 17
             if (process.platform !== 'darwin' || this.architecture !== 'aarch64') {
                 jdkVersions.push({
                     version: [11, 0, 13, 8, 1]
                 });
-                jdkVersions.push({
```
Comment (karianna) [inline dist/setup/index.js:101732]: Seems odd to remove this but to have 11.0.13 remain?

File: src/distributions/microsoft/installer.ts [hunk 4]
```diff
@@ -80,11 +80,17 @@ export class MicrosoftDistributions extends JavaBase {
     // TODO get these dynamically!
     // We will need Microsoft to add an endpoint where we can query for versions.
     const jdkVersions = [
+      {
```
Comment (karianna) [inline src/distributions/microsoft/installer.ts:83]: Same comments as above file

**Review by karianna — CHANGES_REQUESTED** [review]

File: dist/setup/index.js (hunk 1 above)
Comment (gdams) [reply dist/setup/index.js:101721]: It does replace it as the latest. We still preserve `17.0.1.12.1` for history though.

No the version number changed recently to exclude the build number

File: dist/setup/index.js (hunk 3 above)
Comment (gdams) [reply dist/setup/index.js:101732]: I just moved it up

**Review by e-korolevskii — APPROVED** [review]

**Review by dmitry-shibanov — APPROVED** [review]

**Review by marko-zivic-93 — APPROVED** [review]

**Pull request merged**